libs/datasets/timeseries.py

```diff
@@ -1165,8 +1165,9 @@ def write_to_dataset_pointer(self, pointer: dataset_pointer.DatasetPointer):
         """Writes `self` to files referenced by `pointer`."""
         wide_df = self.timeseries_rows()
 
-        # 7 significant digits of precision seems like enough.
-        csv_buf = wide_df.to_csv(index=True, float_format="%.7g")
+        # The number of significant digits needs to be enough that 100M vaccines are formatted as
+        # an integer and not changed to exponential format.
```

Comment:
What is the issue with exponential format? does pandas not handle it properly?

Reply:
Added comment.